Makefile

```diff
@@ -4,24 +4,42 @@ LDFLAGS =
 LDLIBS = -lm
 INCLUDES =
 CFLAGS_COND = -march=native
-
-# Find nvcc
-NVCC := $(shell which nvcc 2>/dev/null)
+SHELL_UNAME = $(shell uname)
+REMOVE_FILES = rm -f
+OUTPUT_FILE = -o $@
 
 # NVCC flags
 NVCC_FLAGS = -O3 --use_fast_math
 NVCC_LDFLAGS = -lcublas -lcublasLt
 
-# Function to test if the compiler accepts a given flag.
-define check_and_add_flag
+ifneq ($(OS), Windows_NT)
+  NVCC := $(shell which nvcc 2>/dev/null)
+
+  # Function to test if the compiler accepts a given flag.
+  define check_and_add_flag
     $(eval FLAG_SUPPORTED := $(shell printf "int main() { return 0; }\n" | $(CC) $(1) -x c - -o /dev/null 2>/dev/null && echo 'yes'))
     ifeq ($(FLAG_SUPPORTED),yes)
         CFLAGS += $(1)
     endif
-endef
+  endef
 
-# Check each flag and add it if supported
-$(foreach flag,$(CFLAGS_COND),$(eval $(call check_and_add_flag,$(flag))))
+  # Check each flag and add it if supported
+  $(foreach flag,$(CFLAGS_COND),$(eval $(call check_and_add_flag,$(flag))))
+else
+  CFLAGS :=
+  REMOVE_FILES = del
+  SHELL_UNAME := Windows
+  NVCC := $(shell where nvcc 2> nul)
+  CC := cl
+  CFLAGS = /Idev /Zi /nologo /Wall /WX- /diagnostics:column /sdl /O2 /Oi /Ot /GL /D _DEBUG /D _CONSOLE /D _UNICODE /D UNICODE /Gm- /EHsc /MD /GS /Gy /fp:fast /Zc:wchar_t /Zc:forScope /Zc:inline /permissive- \
+   /external:W3 /Gd /TP /wd4996 /FC /openmp:llvm
+  LDFLAGS :=
+  LDLIBS :=
+  INCLUDES :=
+  NVCC_FLAGS += -I"dev"
+  WIN_CUDA_RENAME = rename $@.exe $@
+	OUTPUT_FILE = /link /OUT:$@
```

Comment:
spaces/tabs here

Reply:
Got it - will fix that. Having trouble with Makefiles in VS Code with it using the wrong format.